packages/emotion-primitives/src/styled.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+import * as React from 'react'
+import { View, Text, Image } from 'react-primitives'
+import {
+  channel,
+  contextTypes,
+  setTheme,
+  testAlwaysTrue,
+  pickAssign,
+  interleave
+} from './utils'
+import {
+  testPickPropsOnPrimitiveComponent,
+  testPickPropsOnOtherComponent
+} from './test-props'
+import { css } from './css'
+
+function isPrimitiveComponent(component: React.ElementType) {
+  switch (component) {
```

Comment:
is switch case going to be faster than indexOf or object property lookup?

Reply:
Yes. switch cases are realllllllllllllly fast.

Reply:
Object property lookup!

Reply:
We can't do a object property lookup since our keys are functions but switch cases are wayyyyyyyyyyyyyy faster anyway. https://esbench.com/bench/5b3f224bf2949800a0f61e14